react-events-and-state/src/index.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isClicked: false}
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState(prevState => ({
+      isClicked: !prevState.isClicked
+    }));
+  }
+
+
+
+  render() {
+    if (this.state.isClicked) {
+      this.props.text = 'Thank you!';
```

Comment:
`this.props` is read only meaning we should never directly set any properties on it. Here it is better to use a variable:
```JSX
let text = '';

if (this.state.isClicked) {
  text = 'Thank you!';
} else {
  text = 'Click Me!';
}
```